sqlglot/tokens.py

```diff
@@ -1033,15 +1033,19 @@ def _scan_bits(self) -> None:
         self._advance()
         value = self._extract_value()
         try:
-            self._add(TokenType.BIT_STRING, f"{int(value, 2)}")
+            # If `value` can't be converted to a binary, fallback to tokenizing it as an identifier
+            int(value, 2)
+            self._add(TokenType.BIT_STRING, value[2:])  # Drop the 0b
```

Comment:
how do you know the length of 0b?

Reply:
The code path that leads to this function has already checked for the `0b` prefix (same for hex).

Reply:
k sounds good

Reply:
but in theory a dialect could have an arbitrary prefix for byte/hex, and if it's not exactly two characters, this would be wrong - right?

Reply:
The current implementation takes care of `X'..'` and `[0x|0X]...` literals and these changes are robust with respect to this. We can adapt the scanner accordingly in the future if there are any issues, but the current approach should be as robust as it already was (at least it will be with the changes I plan to make soon).

Reply:
Okay, sounds good! Thanks!